docs/rfc/0662-collaborative-objects.adoc

```diff
@@ -0,0 +1,778 @@
+their version of the data at any time. Like so:
+
+[source,typescript]
+----
+import * as Automerge from "automerge"
+
+const data = await fetch("<metadata URL>").then(r => r.json())
+const doc = Automerge.load(base64ToUint8(data.history.changes))
+const updatedDoc = Automerge.change(doc, d => {
+    d.comments.push({
+        "text": "I completely agree!",
+        "author": "<some base64>",
+        "signature": "<some base64>"
+    })
+})
+const change = Automerge.getChanges(doc, updatedDoc)
+const changeBytes = uint8ToBase64(change)
+----
+
+What we do here is load the automerge document from it's history, then use the
+automerge Javscript library to mutate the document (the `Automerge.change`
+call) and then finally get the change between the original version of the 
+document and the new one. 
+
+Now that we have the change we can make a `PATCH` request to 
+`<radicle-implementation>/projects/<project URN>/metadata/<metadata ID>` with
+the following contents:
+
+[source,json]
+----
+{
+    "changes": {
+        "type": "automerge",
+        "change": "<some base64>"
+    }
+}
+----
+
+This endpoint will return an error if the change does not match the schema of
+the object. Otherwise the change will be merged in to the object and announced
+to the network.
+
+== Implementation
+
+Every collaborative object is represented by a graph of automerge changes. See
+the <<appendix_automerge>> for more about automerge. We map this change graph
+to git as described in <<Change Commits>>, we store the tip of the change 
+graph under a `/cob/<typename>/<object ID>` reference, this allows us to 
+reconstruct the state of a collaborative object for a particular peer as 
+described in <<Reconstructing Collaborative Objects>>. This allows us to use
+git to replicate collaborative objects, see <<Fetching Collaborative Objects>>.
+
+We require some schema extensions to authenticate data in collaborative objects,
+these are described in <<Schema Extensions>>.
+
+
+=== Change Commits
+
+Given that automerge changes are a hash linked graph, we can map them to Git.
+We do so by wrapping each change in a commit. The commit points at a tree with
+the following layout
+
+[source]
+----
+.
+|--change
+|--manifest.toml
+----
+
+This tree contains a single change to a collaborative object. We will go into
+more details shortly. Any direct dependencies of this change are encoded in the 
+same manner and become the parents of this commit. This allows us to
+reconstruct the automerge depdency graph. 
+
+Along with the dependencies of the commit we also need to add the commit of the
+identity which created this commit. We need this identity to validate
+signatures and by making the commit a parent we ensure that git will replicate
+it for us. 
+
+A valid change commit must have three trailers:
+
+* `X-Rad-Signature`, as for identity documents
+* `X-Rad-Author-Parent`, this is the hash of the commit which references the
```

Comment:
Is this the root hash or a particular revision of `namespaces/<author>/rad/id`? So in other words, will this hash point to an author at a certain point in time?

Reply:
A particular revision.